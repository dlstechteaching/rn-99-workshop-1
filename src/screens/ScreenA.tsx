import { Image, StyleSheet, View, Text, TouchableOpacity } from 'react-native';
import images from '../assets';
import { Ionicons } from '@expo/vector-icons';
import { StatusBar } from 'expo-status-bar';
import { useNavigation } from '@react-navigation/native';

export const ScreenA: React.VFC = () => {
	const { goBack } = useNavigation();

	return (
		<View style={styles.container}>
			<StatusBar />
			<Image source={images.woman} style={styles.woman} />
			<View style={styles.textContainer}>
				<Text style={styles.title} numberOfLines={2}>
					Never more in rush
				</Text>
				<Text style={styles.subtitle}>
					Check and keep up under control your daily task, is a creative way.
				</Text>
			</View>
			<TouchableOpacity onPress={goBack} style={styles.button}>
				<Ionicons name={'md-arrow-forward'} size={18} color="green" />
			</TouchableOpacity>
		</View>
	);
};

const styles = StyleSheet.create({
	container: {
		flex: 1,
		flexDirection: 'column',
		padding: 30,
		width: '100%',
		backgroundColor: 'white',
		justifyContent: 'space-around',
	},
	woman: {
		marginTop: 40,
		alignSelf: 'flex-start',
		aspectRatio: 2 / 3,
		overflow: 'visible',
		marginLeft: 5,
		marginBottom: -40,
	},
	textContainer: {
		width: '70%',
		justifyContent: 'space-between',
	},
	button: {
		alignSelf: 'flex-end',
		borderRadius: 25,
		flexShrink: 1,
		backgroundColor: 'rgb(236, 248, 248)',
		height: 45,
		width: 45,
		display: 'flex',
		alignItems: 'center',
		justifyContent: 'center',
	},
	title: {
		fontSize: 44,
		marginBottom: 10,
	},
	subtitle: {
		fontSize: 16,
		color: 'rgb(178, 178, 178)',
	},
});
