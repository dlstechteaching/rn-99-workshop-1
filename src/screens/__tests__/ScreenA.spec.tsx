import { useNavigation } from "@react-navigation/native";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import { ScreenA } from "../ScreenA";

jest.mock('@react-navigation/native');

describe("ScreenA tests", () => {
  let component: ShallowWrapper;
  const navigationMock = {
      navigate: jest.fn()
  };

  beforeEach(() => {
    (useNavigation as jest.Mock).mockReturnValue(navigationMock);
  });

  it("should render properly", () => {
    givenComponent();
    thenItRendersProperly();
  });

  function givenComponent() {
    component = shallow(<ScreenA />);
  }

  function thenItRendersProperly() {
    expect(component).toMatchSnapshot();
  }

});