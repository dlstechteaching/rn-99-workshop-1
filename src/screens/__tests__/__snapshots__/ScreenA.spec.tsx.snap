// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenA tests should render properly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-around",
      "padding": 30,
      "width": "100%",
    }
  }
>
  <ExpoStatusBar />
  <Image
    source={1}
    style={
      Object {
        "alignSelf": "flex-start",
        "aspectRatio": 0.6666666666666666,
        "marginBottom": -40,
        "marginLeft": 5,
        "marginTop": 40,
        "overflow": "visible",
      }
    }
  />
  <View
    style={
      Object {
        "justifyContent": "space-between",
        "width": "70%",
      }
    }
  >
    <Text
      numberOfLines={2}
      style={
        Object {
          "fontSize": 44,
          "marginBottom": 10,
        }
      }
    >
      Never more in rush
    </Text>
    <Text
      style={
        Object {
          "color": "rgb(178, 178, 178)",
          "fontSize": 16,
        }
      }
    >
      Check and keep up under control your daily task, is a creative way.
    </Text>
  </View>
  <ForwardRef
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "flex-end",
        "backgroundColor": "rgb(236, 248, 248)",
        "borderRadius": 25,
        "display": "flex",
        "flexShrink": 1,
        "height": 45,
        "justifyContent": "center",
        "width": 45,
      }
    }
  >
    <Icon
      allowFontScaling={false}
      color="green"
      name="md-arrow-forward"
      size={18}
    />
  </ForwardRef>
</View>
`;
